frontend/src/tests/instructor-permission-test.js

```diff
@@ -516,51 +469,254 @@ export function instructorsPermissionTests(api) {
-            (instructor) => instructor.utorid === instructorUser.utorid
-        )?.id;
-        expect(instructorId).toBeDefined();
-        const newDDAH = await createDdahWithFixedDuties(instructorId);
+        beforeEach(async () => {
+            await switchToInstructorOnlyUser();
+        });
 
-        // Test the DDAH is fetched properly
-        await switchToInstructorOnlyUser();
-        resp = await apiGET(`/instructor/sessions/${session.id}/ddahs`);
-        expect(resp).toHaveStatus("success");
-        expect(resp.payload).toHaveLength(1);
-        expect(resp.payload[0]).toStrictEqual(newDDAH);
-    });
+        it("fetch Ddahs associated with self only", async () => {
+            // Our instructor should only be able to fetch ddahs associated with them
+            const resp = await apiGET(
+                `/instructor/sessions/${session.id}/ddahs`
+            );
+            expect(resp).toHaveStatus("success");
+
+            // Only one DDAH is intially seeded for this instructor
+            expect(resp.payload).toHaveLength(1);
+
+            // We get duties returned sorted in ascending order, so
+            // we need to sort the seeded ones before comparing
+            const sortedSeededDuties = databaseSeeder.seededData.ddahs[0].duties.sort(
+                (first, second) => first.order - second.order
+            );
+            expect(resp.payload[0].duties).toEqual(sortedSeededDuties);
+        });
 
-    it.todo("fetch Ddahs a position associated with self");
-    it.todo("fetch Ddahs an assignment associated with self");
-    it.todo("cannot fetch Ddahs for assignment not associated with self");
-    it.todo("create a Ddah for an assignment associated with self");
-    it.todo("update a Ddah for an assignment associated with self");
-    it.todo(
-        "cannot set approved_date/accepted_date/revised_date/emailed_ate/signature for a Ddah associated with self"
-    );
-    it.todo("cannot create a Ddah for an assignment not associated with self");
-    it.todo("cannot update a Ddah for an assignment not associated with self");
+        it("create a Ddah for an assignment associated with self", async () => {
+            const ddahToCreate = {
+                // This assignment has no DDAH initially
+                assignment_id: assignment.id,
+                duties: [
+                    {
+                        order: 1,
+                        hours: 20,
+                        description: "other:Facilitating workshops",
+                    },
+                    {
+                        order: 2,
+                        hours: 50,
+                        description: "other:Lecture support",
+                    },
+                ],
+            };
+            let resp = await apiPOST(`/instructor/ddahs`, ddahToCreate);
+            expect(resp).toHaveStatus("success");
+            expect(resp.payload).toEqual(expect.objectContaining(ddahToCreate));
```

Comment:
How does this work? Isn't `resp.payload` an array? We have a `toContainObject` extension to expect for just this case..

Reply:
In this case `resp.payload` is an object and that was the reason why the test was failing! `toContainObject` would not work with objects.

For `.toEqual` Jest allows its matchers to be passed in the function. It can match a literal or match based on Jest's `expect.matcher()` expressions. In this case we are matching any object that recursively has the properties of the `ddahToCreate`!